### ISSUE TITLE
Enable to schedule Argo CD deployments using workflows

### ISSUE THREAD

**Issue by kivio** [argoproj/argo-workflows#1665]
**Is this a BUG REPORT or FEATURE REQUEST?**:
FEATURE REQUEST
**What happened**:
As user of Argo as whole ecosystem I want to easy schedule Deployment using Workflows that all Argo components can use it.
**What you expected to happen**:
In this moment I need to use Argo CD API to do that using curl container
**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
Idea:
I want simple workflow construction that will schedule deployments.
```yaml
name: schedule-app-deploy
deployment: # it will utilize special container under the hood
  namespace: app-resources
  project: new-frontend
  app: guest-book
```

**Comment by kivio** [comment]
I have done it that way, but as both parts goes from Argo will be nice to simplify it.

**Comment by dcherman** [comment]
This might be solved by the `argocd-sync-and-wait` template available in the Workflow Catalog

https://github.com/argoproj-labs/argo-workflows-catalog

Would that work for your use case?

**Comment by alexec** [comment]
This could be implemented as a plugin too.

**Comment by agilgur5** [comment]
Going to close this out as implemented by plugins per #7912 as well as by [existing templates](https://github.com/argoproj-labs/argo-workflows-catalog/blob/c69f0d992fc0ec5b013d5bdad6bb946047f4c4d7/templates/argocd-sync-and-wait/manifests.yaml#L4) as referenced above
